gateway/ln-gateway/src/federation_manager.rs

```diff
@@ -73,6 +75,24 @@ impl FederationManager {
         Ok(())
     }
 
+    /// Iterates through all of the federations the gateway is registered with
+    /// and requests to remove the registration record.
+    pub async fn leave_all_federations(&self, gateway_keypair: KeyPair) {
```

Comment:
You just copied the name, but it's confusing imo. It's more like "un-announcing" ourselves, but not directly leaving yet (that would imply shutting down the client imo, which this fn can't do with an immutable self ref)

Reply:
Renamed to `unannounce_from_all_federations`